engine/access/rpc/backend/connection_factory.go

```diff
@@ -93,38 +94,52 @@ func (cf *ConnectionFactoryImpl) retrieveConnection(grpcAddress string, timeout
 	return conn, nil
 }
 
-func (cf *ConnectionFactoryImpl) GetAccessAPIClient(address string) (access.AccessAPIClient, io.Closer, error) {
+func (cf *ConnectionFactoryImpl) GetAccessAPIClient(address string) (access.AccessAPIClient, error) {
 
 	grpcAddress, err := getGRPCAddress(address, cf.CollectionGRPCPort)
 	if err != nil {
-		return nil, nil, err
+		return nil, err
 	}
 
 	conn, err := cf.retrieveConnection(grpcAddress, cf.CollectionNodeGRPCTimeout)
 	if err != nil {
-		return nil, nil, err
+		return nil, err
 	}
 
 	accessAPIClient := access.NewAccessAPIClient(conn)
-	closer := io.Closer(conn)
-	return accessAPIClient, closer, nil
+	return accessAPIClient, nil
 }
 
-func (cf *ConnectionFactoryImpl) GetExecutionAPIClient(address string) (execution.ExecutionAPIClient, io.Closer, error) {
+func (cf *ConnectionFactoryImpl) InvalidateAccessAPIClient(address string) bool {
+	grpcAddress, err := getGRPCAddress(address, cf.CollectionGRPCPort)
+	if err != nil {
+		return true
+	}
+	return cf.ConnectionsCache.Remove(grpcAddress)
```

Comment:
we don't have to close the connection before removing it?

Reply:
The logic from here should cover the closing of connections.
https://github.com/onflow/flow-go/blob/c9d7fddceb0ba58b8d537c974d58ace69eb67328/engine/access/rpc/engine.go#L143

This test (https://github.com/onflow/flow-go/blob/c9d7fddceb0ba58b8d537c974d58ace69eb67328/engine/access/rpc/backend/connection_factory_test.go#L387) also checks if our key removal logic would trigger the eviction policy code to close the connection